ts/src/base/Exchange.ts

```diff
@@ -3960,29 +3961,52 @@ export default class Exchange {
         throw new NotSupported (this.id + ' createOrder() is not supported yet');
     }
 
+    async createMarketOrderWithCost (symbol: string, side: OrderSide, cost, params = {}) {
+        /**
+         * @method
+         * @name createMarketOrderWithCost
+         * @description create a market order by providing the symbol, side and cost
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} cost how much you want to trade in units of the quote currency
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        if (this.options['createMarketBuyOrderRequiresPrice'] || this.options['createMarketSellOrderRequiresPrice'] || this.options['createMarketOrderWithCost']) {
+            return await this.createOrder (symbol, 'market', side, cost, 1, params);
```

Comment:
actually I think this might be a bit misleading because the exchange might accept only buy orders with cost

Reply:
so to make it generic this needs to be a &&